### ISSUE TITLE
CMA most likely requires changes to kernel.yama.ptrace_scope

### ISSUE THREAD

**Issue by bertwesarg** [openucx/ucx#3046]
The `process_vm_readv`/`process_vm_writev` syscalls require permissions which are controlled by the `kernel.yama.ptrace_scope` sysctl. Though I could not find any hints to check this value in the documentation. Could you please mention them at least somewhere. Please note, that Open MPI checks the value at runtime, if it uses CMA directly, but not when implicitly using it through UCX.

I have a back trace, where Open MPI uses CMA but failed because of the missing permissions. Unfortunately, Open MPI does not catch this error, I'm still investigating.

```
#0  uct_cma_ep_get_zcopy (tl_ep=0x7728f0, iov=0x7fffffffca80, iovcnt=1, 
    remote_addr=140173892640400, rkey=18446744073709551615, comp=0x7c26d8)
    at sm/cma/cma_ep.c:147
#1  0x00007fffe1f2c073 in uct_ep_get_zcopy (comp=0x7c26d8, 
    rkey=18446744073709551615, remote_addr=140173892640400, iovcnt=1, 
    iov=0x7fffffffca80, ep=0x7728f0)
    at /home/bitten/src/ucx-1.4.0/src/uct/api/uct.h:1833
#2  ucp_rndv_progress_rma_get_zcopy (self=0x7c26f0) at tag/rndv.c:414
#3  0x00007fffe1f2c81a in ucp_request_try_send (req_status=0x7fffffffccf0, 
    req=0x7c2640)
    at /home/bitten/src/ucx-1.4.0/src/ucp/core/ucp_request.inl:168
#4  ucp_request_send (req=0x7c2640)
    at /home/bitten/src/ucx-1.4.0/src/ucp/core/ucp_request.inl:201
#5  ucp_rndv_req_send_rma_get (rndv_req=0x7c2640, rreq=0x7fffffffd2a0, 
    rndv_rts_hdr=0x7fffe8116ac8) at tag/rndv.c:500
#6  0x00007fffe1f2ceea in ucp_rndv_matched (worker=0x7ffff00d7010, 
    rreq=0x7fffffffd2a0, rndv_rts_hdr=0x7fffe8116ac8) at tag/rndv.c:556
#7  0x00007fffe1f2d4c4 in ucp_rndv_process_rts (arg=0x7ffff00d7010, 
    data=0x7fffe8116ac8, length=48, tl_flags=1) at tag/rndv.c:586
#8  0x00007fffe1f2d97d in ucp_rndv_rts_handler (arg=0x7ffff00d7010, 
    data=0x7fffe8116ac8, length=48, tl_flags=1) at tag/rndv.c:610
#9  0x00007fffe186f897 in uct_iface_invoke_am (iface=0x76de60, id=9 '\t', 
    data=0x7fffe8116ac8, length=48, flags=1)
    at /home/bitten/src/ucx-1.4.0/src/uct/base/uct_iface.h:523
#10 0x00007fffe1870193 in uct_mm_iface_invoke_am (flags=1, length=48, 
    data=0x7fffe8116ac8, am_id=9 '\t', iface=0x76de60) at sm/mm/mm_iface.h:127
#11 uct_mm_iface_process_recv (iface=0x76de60, elem=0x7ffff7fee5c0)
    at sm/mm/mm_iface.c:205
#12 0x00007fffe1870395 in uct_mm_iface_poll_fifo (iface=0x76de60)
    at sm/mm/mm_iface.c:239
#13 0x00007fffe1870489 in uct_mm_iface_progress (arg=0x76de60)
    at sm/mm/mm_iface.c:263
#14 0x00007fffe1f0561a in ucs_callbackq_dispatch (cbq=0x762d40)
    at /home/bitten/src/ucx-1.4.0/src/ucs/datastruct/callbackq.h:208
#15 0x00007fffe1f09d30 in uct_worker_progress (worker=0x762d40)
    at /home/bitten/src/ucx-1.4.0/src/uct/api/uct.h:1641
#16 ucp_worker_progress (worker=0x7ffff00d7010) at core/ucp_worker.c:1389
#17 0x00007fffe39e9a1c in mca_pml_ucx_progress ()
    at ../../../../../ompi/mca/pml/ucx/pml_ucx.c:466
#18 0x00007ffff3ac25c4 in opal_progress ()
    at ../../opal/runtime/opal_progress.c:228
#19 0x00007fffe39e9fea in mca_pml_ucx_recv (buf=0x7fffdc1db410, count=2000, 
    datatype=0x604ba0 <ompi_mpi_double>, src=1, tag=10, 
    comm=0x6049a0 <ompi_mpi_comm_world>, mpi_status=0x7fffffffd608)
    at ../../../../../ompi/mca/pml/ucx/pml_ucx.c:551
#20 0x00007ffff526bead in PMPI_Recv (buf=0x7fffdc1db410, count=2000, 
    type=0x604ba0 <ompi_mpi_double>, source=1, tag=10, 
    comm=0x6049a0 <ompi_mpi_comm_world>, status=0x7fffffffd608) at precv.c:79
#21 0x00007ffff7b6339d in MPI_Recv (buf=<optimized out>, count=2000, 
    datatype=0x604ba0 <ompi_mpi_double>, source=1, tag=tag@entry=10, 
    comm=0x6049a0 <ompi_mpi_comm_world>, status=0x7fffffffd608)
    at /home/bitten/builds/835c24de/0/scorepCI/scorep-pipelines/src/build-mpi/../src/adapters/mpi/SCOREP_Mpi_P2p.c:498
#22 0x0000000000401c6e in ExchangeJacobiMpiData (status=0x7fffffffd5c0, 
    request=0x7fffffffd5a0, uold=0x7fffdb299010, data=...)
    at /home/bitten/builds/835c24de/0/scorepCI/scorep-pipelines/src/build-mpi/../test/jacobi/MPI/C++/jacobi.cpp:155
#23 Jacobi (data=...)
    at /home/bitten/builds/835c24de/0/scorepCI/scorep-pipelines/src/build-mpi/../test/jacobi/MPI/C++/jacobi.cpp:66
#24 0x00000000004018c2 in main (argc=1, argv=<optimized out>)
    at /home/bitten/builds/835c24de/0/scorepCI/scorep-pipelines/src/build-mpi/../test/jacobi/MPI/C++/main.cpp:242
```

**Comment by shamisp** [comment]
I think we need to add similar permission check. @yosefe any thoughts ?

**Comment by yosefe** [comment]
@shamisp agree, will add to TODO

**Comment by spikebike** [comment]
This just bit me with a openmpi-4.0.1 + UCX-1.5.1 build.  For anyone reading this, the immediate fix is:
```sudo sysctl -w kernel.yama.ptrace_scope=0```

The permanent fix, at least for Ubuntu 18.04, is to change /etc/sysctl.d/10-ptrace.conf to:
```kernel.yama.ptrace_scope = 0```

The file change will take effect after the next boot.
